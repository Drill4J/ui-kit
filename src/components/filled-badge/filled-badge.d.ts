interface Props {
    className?: string;
    children?: string;
}
export declare const FilledBadge: ({ className, children }: Props) => JSX.Element;
export {};
