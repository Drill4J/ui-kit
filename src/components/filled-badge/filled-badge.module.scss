@import '../../style-guide/index';

.filled-badge {
  border: 1px solid map-get($green, default);
  border-radius: 13px;
  background-color: map-get($green, default);
  color: map-get($monochrome, white);
  font-size: 12px;
  font-family: $semibold-font;
  line-height: 12px;
  padding: 3px 8px;
}
