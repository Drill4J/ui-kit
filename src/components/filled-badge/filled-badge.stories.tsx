import { storiesOf } from '@storybook/react';

import { FilledBadge } from './filled-badge';

storiesOf('FilledBadge', module).add('with text', () => <FilledBadge>Filled badge</FilledBadge>);
