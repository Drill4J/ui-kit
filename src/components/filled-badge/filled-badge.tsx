import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import styles from './filled-badge.module.scss';

interface Props {
  className?: string;
  children?: string;
}

const filledBadge = BEM(styles);

export const FilledBadge = filledBadge(({ className, children }: Props) => (
  <span className={className}>{children}</span>
));
