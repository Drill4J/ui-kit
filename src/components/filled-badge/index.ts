export { FilledBadge } from './filled-badge';
