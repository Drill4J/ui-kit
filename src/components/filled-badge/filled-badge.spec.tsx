import React from 'react';
import renderer from 'react-test-renderer';

import { FilledBadge } from './filled-badge';

it('should match snapshot', () => {
  const tree = renderer.create(<FilledBadge>Filled badge</FilledBadge>).toJSON();

  expect(tree).toMatchSnapshot();
});
