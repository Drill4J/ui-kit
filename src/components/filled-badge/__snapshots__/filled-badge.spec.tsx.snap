// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<span
  className="filled-badge"
>
  Filled badge
</span>
`;
